src/software/ai/navigator/obstacle/obstacle_test.cpp

```diff
@@ -3,231 +3,181 @@
-        EXPECT_NEAR(pg.getPoints()[4].x(), pt4.x(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[4].y(), pt4.y(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[5].x(), pt5.x(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[5].y(), pt5.y(), TEST_EPSILON);
-    }
+    Rectangle rectangle({-1, 1}, {2, -3});
+    Obstacle obstacle(std::make_shared<PolygonObstacle>(PolygonObstacle(rectangle)));
+    Point inside_point(0, -1);
+    Point outside_point(5, 5);
+
+    EXPECT_EQ(obstacle->contains(inside_point), rectangle.contains(inside_point));
+    EXPECT_EQ(obstacle->contains(outside_point), rectangle.contains(outside_point));
 }
 
-// obstacle with robot additional radius buffer = 0, additional velocity projection buffer
-// = 0 centred at (1,2) and with (3,2) velocity
-TEST(NavigatorObstacleTest, shifted_no_buffer_velocity_obstacle_polygon)
+TEST(NavigatorObstacleTest, rectangle_obstacle_distance)
 {
-    double TEST_EPSILON    = 1e-3;
-    Timestamp current_time = Timestamp::fromSeconds(123);
-    Robot robot            = Robot(3, Point(1, 2), Vector(3, 2), Angle::fromRadians(2.2),
-                        AngularVelocity::fromRadians(-0.6), current_time);
-    auto p_pointer = Obstacle::createRobotObstacleWithBufferParams(robot, true, 0.0, 0.0)
-                         .getBoundaryPolygon();
-
-    EXPECT_TRUE(p_pointer);
-
-    if (p_pointer)
-    {
-        Polygon pg = *p_pointer;
-        // visually verified
-        Point pt0 = Point(0.885, 2.173);
-        Point pt1 = Point(.793, 1.987);
-        Point pt2 = Point(.908, 1.814);
-        Point pt3 = Point(1.115, 1.827);
-        Point pt4 = Point(4.115, 3.827);
-        Point pt5 = Point(3.885, 4.173);
-        EXPECT_DOUBLE_EQ(pg.getPoints().size(), 6);
-        EXPECT_NEAR(pg.getPoints()[0].x(), pt0.x(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[0].y(), pt0.y(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[1].x(), pt1.x(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[1].y(), pt1.y(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[2].x(), pt2.x(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[2].y(), pt2.y(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[3].x(), pt3.x(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[3].y(), pt3.y(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[4].x(), pt4.x(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[4].y(), pt4.y(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[5].x(), pt5.x(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[5].y(), pt5.y(), TEST_EPSILON);
-    }
+    Rectangle rectangle({-1, -3}, {2, 1});
+    Obstacle obstacle(std::make_shared<PolygonObstacle>(PolygonObstacle(rectangle)));
+    Point inside_point(0, -1);
+    Point outside_point(5, 5);
+
+    EXPECT_EQ(obstacle->distance(inside_point), ::distance(rectangle, inside_point));
+    EXPECT_EQ(obstacle->distance(outside_point), ::distance(rectangle, outside_point));
```

Comment:
Similar point here, could we test against expected values instead of against other implementations?

Same goes for a lot of the tests below.